DM-40889: Improve developer documentation (#2580)
lsst-sqre/phalanx: docs/developers/deploy-from-a-branch.rst

Normally, these changes will immediately take effect.
Sometimes if you change a ``ConfigMap`` you will need to restart the relevant deployments to pick up that change.
For instructions on how to do that, see :ref:`branch-deploy-restart`.

> (However, see :ref:`dev-deployment-restart` for a better way.)

Reviewer: ```suggestion
(However, see :ref:`dev-deployment-restart` for a better way.)
To configure the chart to make these restarts happen automatically, see :ref:`dev-deployment-restart`.
```

Author: Yes, this is better, but when applying it, I realized that my whole original change was wrong. The checksum is only computed during Helm operations, and this section is on making changes directly in Argo CD, so the restart always has to be done manually. I'll revise it a different way to make that clear.

#. In the application's page in Argo CD, click on the :guilabel:`Sync` button to redeploy the application from the default branch.

> You can also find your application in the ``science-platform`` Argo CD application and sync it from there to reset the default branch and any other settings you changed.
> Either approach works; use whichever you find more convenient.

Reviewer: ```suggestion
Alternatively, you can find the application in the ``science-platform`` Argo CD application and sync it from there to reset the default branch and any other settings you changed.
```
